Algorithm.Python/Alphas/GasAndCrudeOilEnergyCorrelationAlpha.py

```diff
@@ -0,0 +1,242 @@
+'''
+    Energy prices, especially Oil and Natural Gas, are in general fairly correlated,
+    meaning they typically move in the same direction as an overall trend. This Alpha
+    uses this idea and implements an Alpha Model that takes Natural Gas ETF price
+    movements as a leading indicator for Crude Oil ETF price movements. We take the
+    Natural Gas/Crude Oil ETF pair with the highest historical price correlation and
+    then create insights for Crude Oil depending on whether or not the Natural Gas ETF price change
+    is above/below a certain threshold that we set (arbitrarily).
+    
+    
+    
+    This alpha is part of the Benchmark Alpha Series created by QuantConnect which are open
+    sourced so the community and client funds can see an example of an alpha.
+'''
+
+
+
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Indicators import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Algorithm.Framework import *
+from QuantConnect.Algorithm.Framework.Alphas import *
+from QuantConnect.Algorithm.Framework.Execution import *
+from QuantConnect.Algorithm.Framework.Portfolio import PortfolioTarget, EqualWeightingPortfolioConstructionModel
+from Portfolio import *
+from Execution import *
+from Risk.NullRiskManagementModel import NullRiskManagementModel
+from QuantConnect.Algorithm.Framework.Selection import * 
+
+import numpy as np
+from scipy import stats
+from scipy.stats import kendalltau 
+from datetime import timedelta, datetime
+
+class EnergyETFPairsTradingAlpha(QCAlgorithmFramework):
+
+    def Initialize(self):
+        
+        ## Pick 3 natural gas ETFs and 3 crude oil ETFs and use all 9 possible combinations
+        ## One combination of natural gas ETF/crude oil ETF will be used
+        self.tickers = [["UNG","UNG","UNG",'BOIL','BOIL','BOIL','UNL','UNL','UNL'],   ## Natural gas ETFs
+                        ["USO",'UCO','DBO',"USO",'UCO','DBO',"USO",'UCO','DBO']]      ## Crude oil ETFs
+        tickers = ['UNG','BOIL','UNL','USO','UCO','DBO']
+        self.SetStartDate(2018, 1, 1)   #Set Start Date
+        self.SetCash(100000)           #Set Strategy Cash
+        
+        ## Set Universe resolution and subscribed to data
+        self.Universe.Resolution = Resolution.Minute
+        symbols = [ Symbol.Create(ticker, SecurityType.Equity, Market.USA) for ticker in tickers ]
+
+        ## Manual Universe Selection
+        self.SetUniverseSelection( ManualUniverseSelectionModel(symbols) )
+        
+        ## Custom Alpha Model
+        self.SetAlpha(PairsAlphaModel(pairs_tickers = self.tickers, history_days = 90))
```

Comment:
Just a comment (no changes required).
We could pass two lists: `natural_gas` and `crude_oil`:
```python
natural_gas = ['UNG','BOIL','UNL']
crude_oil = ['USO','UCO','DBO']
symbols = [ Symbol.Create(ticker, SecurityType.Equity, Market.USA) for ticker in natural_gas + crude_oil ]
# ...
self.SetAlpha(PairsAlphaModel(leading = natural_gas, following = crude_oil, history_days = 90))
```
it would improve readability and make some operations easier in the alpha model.

Reply:
💯 great idea